webapp/src/js/services/enketo.js

```diff
@@ -349,7 +356,28 @@ angular.module('inboxServices').service('Enketo',
         $(selector).on('valuechange.enketo', listener);
       }
     };
-                        
+
+    var registerValidationCompleteListener = function(selector) {
+      const listener = () => {
+        if ($('.invalid-required').length > 0) {
```

Comment:
This is searching the entire DOM for this class. I think it'd be better to restrict it to just: `$(selector).find('.invalid-required')...`.

Then because you're using it below you should pull out a variable, eg: `const $selector = $(selector);`